inst/examples/sir2.cpp

```diff
@@ -0,0 +1,123 @@
+double ldpois(double x, double lambda) {
+  return x * std::log(lambda) - lambda - std::lgamma(x + 1);
+}
+
+class sir2 {
+public:
+  typedef double real_t;
+  struct init_t {
+    double S0;
+    double I0;
+    double R0;
+    double beta;
+    double gamma;
+    double dt;
+    size_t freq;
+    // Observation parameters
+    double exp_noise;
+  };
+  // Presence of this is going to be important
+  struct data_t {
+    double incidence;
+  };
```

Comment:
It's not static because it ends up in `std::map<size_t, data_t>` with each timestep having a different `data_t` with a different incidence value

This example could have been simplified further by doing `typedef double data_t` but I wanted to show creating a struct here as that's the most likely solution